homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,221 @@
+            )
+            image_type = info.get(CONF_CONTENT_TYPE)
+            if image_type is not None and image_type not in ["image/jpeg", "image/png"]:
+                _LOGGER.debug(
+                    "Ignoring image format check because user specified a format"
+                )
+                user_format = info.get(CONF_CONTENT_TYPE)
+                if fmt is None or fmt not in user_format:
+                    _LOGGER.warning(
+                        "Still image at '%s' detected format: '%s' does not match user specified format: '%s'",
+                        info[CONF_STILL_IMAGE_URL],
+                        fmt,
+                        user_format,
+                    )
+            elif fmt is None:
+                return False, "invalid_still_image"
+
+        # Second level functionality is to get a stream.
+        if info.get(CONF_STREAM_SOURCE) in [None, ""]:
+            return True, ""
+        stream_source = await cam.stream_source()
+        try:
+            # For RTSP streams, prefer TCP. This code is duplicated from
+            # homeassistant.components.stream.__init__.py:create_stream()
+            # It may be possible & better to call create_stream() directly.
+            stream_options: dict[str, str] = {}
+            if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                stream_options = {
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            _LOGGER.debug("Attempting to open stream %s", stream_source)
+            container = await self.hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+            return True, ""
+        # pylint: disable=c-extension-no-member
+        except av.error.FileNotFoundError:
+            return False, "stream_no_route_to_host"
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return False, "stream_unauthorised"
+        except (KeyError, IndexError):
+            return False, "stream_novideo"
+        except PermissionError:
+            return False, "stream_not_permitted"
+        except OSError as err:
+            if "No route to host" in str(err):
+                return False, "stream_no_route_to_host"
+            if "Input/output error" in str(err):
+                return False, "stream_io_error"
+            raise err
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
```

Comment:
We should auto generate a name based on the other inputs, set it as `entry.title` and use `entry.title` for the name. They can rename it later

Reply:
My plan was to let the user type a name at the end of the config flow once I've added an an image preview (separate PR).  Any automatic name/title would be derived from the URL (quite often an IP address) so I think it would not be a friendly name.

Reply:
Now done using the URL as the name, which is then editable in the options flow.  So I think this is done.